src/errors/error_handler.js

```diff
@@ -33,34 +33,34 @@ export default class ErrorHandler {
     }
 
     if (statusCode >= 300 && statusCode < 400) {
-      throw new RedirectError(errorParams);
+      return new RedirectError(errorParams);
     }
 
     switch (statusCode) {
       case 401:
-        throw new UnauthorizedError(errorParams);
+        return new UnauthorizedError(errorParams);
```

Comment:
Question: How did the test suite not catch this?

Reply:
Because it wasn't necessarily wrong. But we want this function to simply construct an exception, and it will be thrown upstream, rather than thrown inside this function